extensions/kubernetes-client/deployment-internal/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesResourceBuildStep.java

```diff
@@ -14,19 +20,42 @@
 import io.quarkus.deployment.annotations.Record;
 import io.quarkus.kubernetes.client.runtime.KubernetesResources;
 import io.quarkus.kubernetes.client.runtime.KubernetesSerializationRecorder;
+import io.quarkus.kubernetes.client.spi.KubernetesResourcesBuildItem;
 
 public class KubernetesResourceBuildStep {
+
+    @BuildStep
+    void scanKubernetesResourceClasses(BuildProducer<KubernetesResourcesBuildItem> kubernetesResourcesBuildItemBuildProducer) {
+        final var classLoaders = new ClassLoader[] {
+                Thread.currentThread().getContextClassLoader(),
+                KubernetesDeserializer.class.getClassLoader() };
```

Comment:
Do we really need both ClassLoaders?

Reply:
I have no idea if this is needed at build time (or even in Quarkus). It is used as a workaround and fallback mechanism in Kubernetes Client, related to several issues with groovy and others.

Reply:
I would try with only the TCCL and see if that works (it should) and get rid of the complexity if so

Reply:
I guess it should work, and having moved the loading to build time, runtime shouldn't be the issue.

Reply:
changed